### PR TITLE
Branch dg diagram

### PR DESCRIPTION
**Pull request by Justinhoejj**
Add diagrams to DG.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -263,12 +263,15 @@ Step 8. Contact's `Pin` attribute will change to indicate that the contact is no
 
 Step 9. CONNECTIONS UI will update to show the contact behind other pinned contacts using a `PersonCard`. 
 
-The following sequence diagram shows how the pin operation works:
+The following sequence diagram shows how the `pin operation works:
```
Comment (cookiedan42) [inline docs/DeveloperGuide.md:266]: missing closing `?

File: docs/diagrams/BirthdayReminderSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+CommandResult --> AddCommand : r
+deactivate CommandResult
+
+[<-- AddCommand : r
+deactivate AddCommand
+
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+[<-[hidden]- AddCommand
+destroy AddCommand
+@enduml
```
Comment (cookiedan42) [inline docs/diagrams/BirthdayReminderSequenceDiagram.puml:72]: EOF newline

**Review by cookiedan42 — APPROVED** [review]
Add your small missing stuff :)

**Review by leezhixuan — APPROVED** [review]
LGTM

**Pull request merged**